anthos-bm-gcp-bash/install_admin_cluster.sh

```diff
@@ -44,286 +44,357 @@ fi
-  --role="roles/monitoring.dashboardEditor" \
-  --no-user-output-enabled
-
-gcloud projects add-iam-policy-binding "$PROJECT_ID" \
-  --member="serviceAccount:baremetal-gcr@$PROJECT_ID.iam.gserviceaccount.com" \
-  --role="roles/stackdriver.resourceMetadata.writer" \
-  --no-user-output-enabled
-
-gcloud projects add-iam-policy-binding "$PROJECT_ID" \
-  --member="serviceAccount:baremetal-gcr@$PROJECT_ID.iam.gserviceaccount.com" \
-  --role="roles/opsconfigmonitoring.resourceMetadata.writer" \
-  --no-user-output-enabled
-# [END anthos_bm_gcp_bash_admin_add_iam_role]
-printf "✅ Successfully added the requires IAM roles to the Service Account.\n\n"
-
-# declare arrays for VM names and IPs
-printf "🔄 Setting up array variables for the VM names and IP addresses...\n"
-# [START anthos_bm_gcp_bash_admin_vms_array]
-MACHINE_TYPE=n1-standard-8
-VM_PREFIX=abm
-VM_WS=$VM_PREFIX-ws
-VM_ADMIN_CP=$VM_PREFIX-admin-cluster-cp
-VM_USER_CP=$VM_PREFIX-user-cluster-cp
-VM_USER_W1=$VM_PREFIX-user-cluster-w1
-VM_USER_W2=$VM_PREFIX-user-cluster-w2
-declare -a VMs=("$VM_WS" "$VM_ADMIN_CP" "$VM_USER_CP" "$VM_USER_W1" "$VM_USER_W2")
-declare -a IPs=()
-# [END anthos_bm_gcp_bash_admin_vms_array]
-printf "✅ Variables for the VM names and IP addresses setup.\n\n"
-
-# create GCE VMs
-printf "🔄 Creating GCE VMs...\n"
-# [START anthos_bm_gcp_bash_admin_create_vm]
-for vm in "${VMs[@]}"
+# --------------------------------------------------------------
+# This section checks what portions of the script the user wants
+# to execute. The script does two things. (1) Set up the GCE
+# infrastrcuture where the Anthos bare metal cluster will be
+# installed and (2) Install the Anthos bare metal cluster. Thus,
+# in this section we provide the user with the option to do run
+# everything or to only one of these two things.
+# --------------------------------------------------------------
+
+RUN_ALL="All-(Setup-and-Install)"
+SETUP_ONLY="Setup-Only"
+INSTALL_ONLY="Install-Only"
+
+PS3=$'\n''Please select an installation mode: '
+options=("$RUN_ALL" "$SETUP_ONLY" "$INSTALL_ONLY" "Quit")
+select OPT in "${options[@]}"
 do
-    gcloud compute instances create "$vm" \
-      --image-family=ubuntu-2004-lts --image-project=ubuntu-os-cloud \
-      --zone="${ZONE}" \
-      --boot-disk-size 200G \
-      --boot-disk-type pd-ssd \
-      --can-ip-forward \
-      --network default \
-      --tags http-server,https-server \
-      --min-cpu-platform "Intel Haswell" \
-      --scopes cloud-platform \
-      --machine-type "$MACHINE_TYPE" \
-      --metadata "cluster_id=${ADMIN_CLUSTER_NAME},bmctl_version=${BMCTL_VERSION}"
-    IP=$(gcloud compute instances describe "$vm" --zone "${ZONE}" \
-         --format='get(networkInterfaces[0].networkIP)')
-    IPs+=("$IP")
+  case $OPT in
+    "$RUN_ALL")
+      printf "\nYou chose '%s'." "$OPT"
+      printf "This will setup the GCE infrastructure and install the Anthos bare metal cluster."
```

Comment:
Clarify that the admin cluster will be installed.

Reply:
Fixed